app/views/_create-secret.html

```diff
@@ -0,0 +1,267 @@
+<!-- Error messages -->
+<parse-error error="error" ng-if="error"></parse-error>
+
+<form name="form">
+
+  <div class="form-group">
+    <label>Secret Type</label>
+    <ui-select required ng-model="newSecret.type" search-enabled="false" ng-change="newSecret.authType = secretAuthTypeMap[newSecret.type][0]">
```

Comment:
If you click "Create source secret", you shouldn't be able to change the secret type in the modal.